symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java

```diff
@@ -364,55 +426,85 @@ public V4Message getMessage(@Nonnull String messageId) {
     return executeAndRetry("getMessageIdsByTimestamp", () ->
         defaultApi.v2AdminStreamsStreamIdMessageIdsGet(authSession.getSessionToken(), streamId, getEpochMillis(since),
-            getEpochMillis(to), limit, skip));
+            getEpochMillis(to), null, null));
   }
 
   /**
-   * Fetches message ids using timestamp.
+   * Fetches message ids using timestamp and returns in {@link Stream}.
    *
-   * @param streamId  the ID of the stream where to fetch messages.
-   * @param since     optional instant of the first required messageId.
-   * @param to        optional instant of the last required messageId.
-   * @param chunkSize size of elements to retrieve in one call. Optional and defaults to 50.
-   * @param totalSize total maximum number of messages to return. Optional and defaults to 50.
+   * @param streamId   the ID of the stream where to fetch messages.
+   * @param since      optional instant of the first required messageId.
+   * @param to         optional instant of the last required messageId.
+   * @param pagination The chunkSize and totalSize for pagination.
    * @return a {@link Stream} containing the messageIds.
    * @see <a href="https://developers.symphony.com/restapi/reference#get-message-ids-by-timestamp">Get Message IDs by Timestamp</a>
    */
-  public Stream<String> getMessageIdsByTimestampStream(@Nonnull String streamId, Instant since, Instant to,
-      Integer chunkSize, Integer totalSize) {
+  @API(status = API.Status.EXPERIMENTAL)
+  public Stream<String> listAllMessageIdsByTimestampStream(@Nonnull String streamId, @Nullable Instant since,
+      @Nullable Instant to, @Nonnull StreamPaginationAttribute pagination) {
     PaginatedApi<String> api = ((offset, limit) ->
-        getMessageIdsByTimestamp(streamId, since, to, limit, offset).getData());
+        listMessageIdsByTimestamp(streamId, since, to, new PaginationAttribute(offset, limit)).getData());
 
-    final int actualChunkSize = chunkSize == null ? 50 : chunkSize.intValue();
-    final int actualTotalSize = totalSize == null ? 50 : totalSize.intValue();
-
-    return new PaginatedService<>(api, actualChunkSize, actualTotalSize).stream();
+    return new PaginatedService<>(api, pagination.getChunkSize(), pagination.getTotalSize()).stream();
   }
 
+  /**
+   * Fetches message ids using timestamp and returns in {@link Stream} with default chunk size and total size equals 100.
+   *
+   * @param streamId the ID of the stream where to fetch messages.
+   * @param since    optional instant of the first required messageId.
+   * @param to       optional instant of the last required messageId.
+   * @return a {@link Stream} containing the messageIds.
+   * @see <a href="https://developers.symphony.com/restapi/reference#get-message-ids-by-timestamp">Get Message IDs by Timestamp</a>
+   */
+  @API(status = API.Status.EXPERIMENTAL)
+  public Stream<String> listAllMessageIdsByTimestampStream(@Nonnull String streamId, @Nullable Instant since,
```

Comment:
Stil `Stream` at the end?

Reply:
I'm not sure about that. That's why I'm still annotated it as `EXPERIMENTAL`

Reply:
_listAllMessageIdsByTimestamp_**Stream**() :)

Reply:
Oops 😅